cmd/azure.go

```diff
@@ -41,6 +45,15 @@ type AzureObjects struct {
 
 // Convert azure errors to minio object layer errors.
 func azureToObjectError(err error, params ...string) error {
+	if err == nil {
+		return nil
+	}
+
+	e, ok := err.(*Error)
+	if ok {
+		err = e.e
```

Comment:
this way it loses stacktrace context if err is non-azure `*Error`, for ex. error returned by `azureParseBlockID()`, hence the previous style.